src/Core/Utility.js

```diff
@@ -156,3 +156,20 @@ function each (iterable, callback) {
 function invariant (from, message) {
 	throw new Error('#'+from+'(...): '+message+'.')
 }
+
+/**
+ * @param {Object} a
+ * @param {Object} b
+ * @return {boolean}
+ */
+function compare (a, b) {
+  for (var i in a)
```

Comment:
Would it be faster to compare `Object.keys(a).length` instead of iterating keys twice?

[React implementation](https://github.com/facebook/fbjs/blob/0c03b9c3788eaf19ec19598a00f319a5f0d831f2/packages/fbjs/src/core/shallowEqual.js#L41) (handles +0/-0, and NaN)

Reply:
Not sure, might depend on the number of key-value pairs. `for in` would do better when that number is lower(which is the common case) while Object.keys() would yield better results with a higher number.